core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -140,51 +139,61 @@ public DockerClient client() {
                 "  Operating System: " + dockerInfo.getOperatingSystem() + "\n" +
                 "  Total Memory: " + dockerInfo.getMemTotal() / (1024 * 1024) + " MB");
 
-        String ryukContainerId = null;
+        final String ryukContainerId;
+
         boolean useRyuk = !Boolean.parseBoolean(System.getenv("TESTCONTAINERS_RYUK_DISABLED"));
         if (useRyuk) {
+            log.debug("Ryuk is enabled");
             ryukContainerId = ResourceReaper.start(hostIpAddress, client);
             log.info("Ryuk started - will monitor and terminate Testcontainers containers on JVM exit");
+        } else {
+            log.debug("Ryuk is disabled");
+            ryukContainerId = null;
         }
 
         boolean checksEnabled = !TestcontainersConfiguration.getInstance().isDisableChecks();
         if (checksEnabled) {
-            VisibleAssertions.info("Checking the system...");
-            checkDockerVersion(version.getVersion());
-            if (ryukContainerId != null) {
-                checkDiskSpace(client, ryukContainerId);
-            } else {
-                runInsideDocker(
-                    client,
-                    createContainerCmd -> {
-                        createContainerCmd.withName("testcontainers-checks-" + SESSION_ID);
-                        createContainerCmd.getHostConfig().withAutoRemove(true);
-                        createContainerCmd.withCmd("tail", "-f", "/dev/null");
-                    },
-                    (__, containerId) -> {
-                        checkDiskSpace(client, containerId);
-                        return "";
-                    }
-                );
+            log.debug("Checks are enabled");
+            // fail-fast if checks have failed previously
+            if (cachedChecksFailure != null) {
+                log.debug("There is a cached checks failure - throwing", cachedChecksFailure);
+                throw cachedChecksFailure;
```

Comment:
it seems that we're throwing too late. We should do it before attempting to start Ryuk, otherwise the method will fail due to Ryuk name conflict and not the cached failure